dcicutils/env_utils.py

```diff
@@ -47,3 +99,48 @@ def is_stg_or_prd_env(envname):
     elif any(token in envname for token in stg_or_prd_tokens):
         return True
     return False
+
+
+def is_test_env(envname):
+    return envname in BEANSTALK_TEST_ENVS
+
+
+def is_hotseat_env(envname):
+    return 'hot' in envname
+
+
+def get_env_from_context(settings, allow_environ=False):
+    if allow_environ:
+        environ_env_name = os.environ.get('ENV_NAME')
+        if environ_env_name:
+            return environ_env_name
+    return settings.get('env.name')
+
+
+def get_mirror_env_from_context(settings, allow_environ=False, allow_guess=True, ):
+    # TODO: I added allow_environ featurism here but did not yet enable it.
+    #       Want to talk to Will about whether we should consider that a compatible or breaking hcange.
```

Comment:
This sounds fine to me, it would be nice though if we could enable allow_environ with an environment variable (I know that sounds weird) because we could then pull settings from Elastic Beanstalk templates.

Reply:
`allow_environ` only turns on whether it looks for the environment variable. If the environment variable is not there, then it acts as if you want it to use declaration files or other means. The order is:
* Look at environment variable for a mirror environment (if allow_environ is True)
* Look in config.
* Look for an environment variable for regular env and guess mirror based on that (if allow_environ is True and allow_guess is True)
* Look for regular env in config and guess mirror based on that (if allow_environ is True and allow_guess is True)
If you want all four tests to be done that way, I would be OK with that.

Reply:
Makes sense.